lms/djangoapps/ccx/migrations/0006_set_display_name_as_override.py

```diff
@@ -0,0 +1,155 @@
+        )
+
+        for override in query:
+            block_overrides = overrides.setdefault(override.location, {})
+            block_overrides[override.field] = json.loads(override.value)
+            block_overrides[override.field + "_id"] = override.id
+            block_overrides[override.field + "_instance"] = override
+
+        overrides_cache[ccx] = overrides
+
+    return overrides_cache[ccx]
+
+
+@transaction.atomic
+def override_field_for_ccx(ccx, block, name, value, CcxFieldOverride):
+    """
+    Overrides a field for the `ccx`.  `block` and `name` specify the block
+    and the name of the field on that block to override.  `value` is the
+    value to set for the given field.
+    """
+    field = block.fields[name]
+    value_json = field.to_json(value)
+    serialized_value = json.dumps(value_json)
+    override_has_changes = False
+    clean_ccx_key = _clean_ccx_key(block.location)
+
+    override = get_override_for_ccx(ccx, block, name + "_instance", CcxFieldOverride)
+    if override:
+        override_has_changes = serialized_value != override.value
+
+    if not override:
+        override, created = CcxFieldOverride.objects.get_or_create(
+            ccx=ccx,
+            location=block.location,
+            field=name,
+            defaults={'value': serialized_value},
+        )
+        if created:
+            _get_overrides_for_ccx(ccx, CcxFieldOverride).setdefault(clean_ccx_key, {})[name + "_id"] = override.id
+        else:
+            override_has_changes = serialized_value != override.value
+
+    if override_has_changes:
+        override.value = serialized_value
+        override.save()
+
+    _get_overrides_for_ccx(ccx, CcxFieldOverride).setdefault(clean_ccx_key, {})[name] = value_json
+    _get_overrides_for_ccx(ccx, CcxFieldOverride).setdefault(clean_ccx_key, {})[name + "_instance"] = override
+
+
+def save_display_name(apps, schema_editor):
+    # We can't import the Person model directly as it may be a newer
+    # version than this migration expects. We use the historical version.
+    CcxFieldOverride = apps.get_model('ccx', 'CcxFieldOverride')
+    CustomCourseForEdX = apps.get_model('ccx', 'CustomCourseForEdX')
+    ccx_display_name_present_ids = list(CcxFieldOverride.objects.filter(
+        field='display_name').values_list('ccx__id', flat=True))
+    ccx_list = CustomCourseForEdX.objects.exclude(id__in=ccx_display_name_present_ids)
+    for ccx in ccx_list:
+        course = get_course_by_id(ccx.course_id, depth=None)
+        override_field_for_ccx(ccx, course, 'display_name', ccx.display_name, CcxFieldOverride)
+        # using CCX object as sender here.
+        responses = SignalHandler.course_published.send(
+            sender=ccx,
+            course_key=CCXLocator.from_course_locator(course.id, unicode(ccx.id))
+        )
+        for rec, response in responses:
+            log.info('Signal fired when course is published. Receiver: %s. Response: %s', rec, response)
```

Comment:
Please specify the course that's being published.

Reply:
This is fixed.